File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
        Arrays.sort(phone_book);
        int size = phone_book.length;
        
        for (int i = 0; i < size - 1; i++) {
            if (phone_book[i+1].startsWith(phone_book[i])) {
                return false;
            } 
        }
        return true;
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    answer = True
    new = sorted(phone_book)

    for i in range(len(new)-1):
        
        if len(new[i]) < len(new[i+1]):

            if new[i] == new[i+1][:len(new[i])]:
                answer = False
                break

    return answer